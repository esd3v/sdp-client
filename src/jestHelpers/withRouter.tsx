import * as React from 'react';
import {Router} from 'react-router-dom';
import {createBrowserHistory} from 'history';

const customHistory = createBrowserHistory();

export const withRouter = (Component: React.FunctionComponent) => (
  <Router history={customHistory}>
    <Component/>
  </Router>
);
