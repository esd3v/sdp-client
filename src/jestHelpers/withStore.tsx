import * as React from 'react';
import {Provider} from 'react-redux';
import {store} from 'store';

export const withStore = (Component: React.FunctionComponent) => (
  <Provider store={store}>
      <Component/>
  </Provider>
);
