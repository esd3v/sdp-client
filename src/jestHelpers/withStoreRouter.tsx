import * as React from 'react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createBrowserHistory} from 'history';
import {store} from 'store';

const customHistory = createBrowserHistory();

export const withStoreRouter = (Component: React.FunctionComponent) => (
  <Provider store={store}>
    <Router history={customHistory}>
      <Component/>
    </Router>
  </Provider>
);
