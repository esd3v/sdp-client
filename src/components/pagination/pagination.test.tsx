import React from 'react';
import {Pagination} from './';
import renderer from 'react-test-renderer';
import {withStoreRouter} from 'jestHelpers';

test('renders correctly', () => {
  const tree = renderer
    .create(withStoreRouter({
      component: <Pagination onSwitch={() => {}}/>,
    }))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
