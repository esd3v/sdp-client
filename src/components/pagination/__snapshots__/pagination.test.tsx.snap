// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  .emotion-0 {
  color: #fff;
  font-size: 16px;
  background-color: #448AFF;
  box-shadow: 0 1px 3px rgba(0,0,0,0.24),0 1px 2px rgba(0,0,0,0.36);
  min-width: 64px;
  border-radius: 3px;
}

.emotion-0:hover:enabled {
  background-color: #6aa2ff;
}

.emotion-0:active:enabled {
  background-color: #1e72ff;
}

.emotion-0:disabled {
  background-color: #bfbfbf;
}

.emotion-5 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0;
}

.emotion-1 {
  padding: 6px 8px;
}

.emotion-2 {
  width: 50%;
  margin: 0 16px;
}

<div>
    <div
      className="emotion-5"
    >
      <button
        className="emotion-0 emotion-1"
        disabled={false}
        onClick={[Function]}
      >
        Prev
      </button>
      <select
        className="emotion-2"
        onChange={[Function]}
        value={0}
      />
      <button
        className="emotion-0 emotion-1"
        disabled={false}
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>,
  .emotion-2 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0;
}

.emotion-0 {
  width: 32px;
  height: 100%;
  cursor: pointer;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 1px 3px rgba(0,0,0,0.24),0 1px 2px rgba(0,0,0,0.36);
  background-color: #fff;
  width: auto;
  height: auto;
  color: initial;
  font-size: 16px;
}

.emotion-0:hover {
  color: #fff;
  background-color: #448AFF;
}

.emotion-0 a {
  padding: 6px 8px;
}

<div>
    <ul
      className="emotion-2"
    >
      <li
        className="emotion-0 disabled"
      >
        <a
          aria-disabled="true"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Prev
        </a>
      </li>
      <li
        className="emotion-0"
      >
        <a
          aria-disabled="false"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Next
        </a>
      </li>
    </ul>
  </div>,
]
`;
